### PR TITLE
List EKS clusters endpoint

### LINKED ISSUE #8033: Create EKS cluster list endpoint
**User Story**

**Acceptance criteria**

### PR DESCRIPTION
**Pull request by imharshita**
Signed-off-by: Harshita <harshita.sharma6174@gmail.com>

**What this PR does / why we need it**:
Adds EKS cluster List endpoint

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #8033

**Does this PR introduce a user-facing change?**:

```release-note
Adds GET /api/v2/providers/eks/clusters
```

### CODE REVIEW

File: pkg/crd/kubermatic/v1/external_cluster.go [hunk 1]
```diff
@@ -68,13 +68,20 @@ type ExternalClusterSpec struct {
...
 }
 
 type ExternalClusterGKECloudSpec struct {
 	Name                 string                                  `json:"name"`
 	CredentialsReference *providerconfig.GlobalSecretKeySelector `json:"credentialsReference,omitempty"`
 }
 
+type ExternalClusterEKSCloudSpec struct {
+	Name                 string                                  `json:"name"`
+	CredentialsReference *providerconfig.GlobalSecretKeySelector `json:"credentialsReference,omitempty"`
```
Comment (mfranczy) [inline pkg/crd/kubermatic/v1/external_cluster.go]: is that optional parameter? If yes could you move it below required fields?

**Review by mfranczy — COMMENTED** [review]
Just a nit, let's wait for @zreigz review

File: pkg/crd/kubermatic/v1/external_cluster.go (hunk 1 above)
Comment (zreigz) [reply pkg/crd/kubermatic/v1/external_cluster.go]: this is mandatory to store credentials in the secret for the specific provider

Comment (mfranczy) [reply pkg/crd/kubermatic/v1/external_cluster.go]: then could we remove `omitempty`?

Comment (imharshita) [reply pkg/crd/kubermatic/v1/external_cluster.go]: yes, we can remove it. wdyt? @zreigz

Comment (zreigz) [reply pkg/crd/kubermatic/v1/external_cluster.go]: yes, let's remove it

**Pull request merged**